packages/form/src/form/VirtualizedAutocompleteList/VirtualizedAutocompleteList.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { VariableSizeList, ListChildComponentProps } from 'react-window'
+import get from 'lodash/get'
+import { useResetCache } from './hooks/useResetCache'
+import {
+  CHARACTERS_PER_LINE,
+  LISTBOX_PADDING,
+  MAX_VISIBLE_ITEM_COUNT,
+  VIRTUALIZED_LIST_ITEM_SIZE,
+} from './constants'
+import { AutocompleteListDataItem } from '../fields/types'
+import { renderOptionFromListDataItem } from '../utils/renderModelFieldOption'
+
+// for MUI Autocomplete to pass renderOption props to
+// this virtualized list' renderOption function
+const OuterElementContext = React.createContext({})
+const OuterElementType = React.forwardRef<HTMLDivElement>((props, ref) => {
+  const outerProps = React.useContext(OuterElementContext)
+  return <div ref={ref} {...props} {...outerProps} />
+})
+
+const renderOption = (optionProps: ListChildComponentProps) => {
+  const { data, index, style } = optionProps
+  const dataSet = data[index]
+  const props = dataSet[0] ?? {}
+  const inlineStyle = {
+    ...style,
+    top: (style.top as number) + LISTBOX_PADDING,
+  }
+
+  return (
+    <li {...props} style={inlineStyle}>
+      {renderOptionFromListDataItem(dataSet)}
+    </li>
+  )
+}
+
+export const VirtualizedAutocompleteList = React.forwardRef<
```

Comment:
Move to form/fields/ModelField instead? Prefer to colocate both components since they are pretty coupled